### PR TITLE
Update collective ops docs

### PR DESCRIPTION
**Pull request by HermitSun**
更新集合通信操作的中文文档，包括：
- paddle.distributed.all_gather
- paddle.distributed.all_reduce
- paddle.distributed.alltoall
- paddle.distributed.broadcast
- paddle.distributed.reduce
- paddle.distributed.recv
- paddle.distributed.irecv
- paddle.distributed.send
- paddle.distributed.isend
- paddle.distributed.scatter
- paddle.distributed.reduce_scatter
- paddle.distributed.new_group

增加集合通信操作的中文文档，包括：
- paddle.distributed.destroy_process_group
- paddle.distributed.alltoall_single
- paddle.distributed.stream.all_gather
- paddle.distributed.stream.all_reduce
- paddle.distributed.stream.alltoall
- paddle.distributed.stream.alltoall_single
- paddle.distributed.stream.broadcast
- paddle.distributed.stream.recv
- paddle.distributed.stream.reduce
- paddle.distributed.stream.reduce_scatter
- paddle.distributed.stream.send
- paddle.distributed.stream.scatter

实现PR包括：
- https://github.com/PaddlePaddle/Paddle/pull/45574
- https://github.com/PaddlePaddle/Paddle/pull/45844
- https://github.com/PaddlePaddle/Paddle/pull/45282
- https://github.com/PaddlePaddle/Paddle/pull/46023
- https://github.com/PaddlePaddle/Paddle/pull/46295
- https://github.com/PaddlePaddle/Paddle/pull/46761
- https://github.com/PaddlePaddle/Paddle/pull/46962
- https://github.com/PaddlePaddle/Paddle/pull/46922

PADDLEPADDLE_PR=46922

### CODE REVIEW

File: docs/api/paddle/distributed/reduce_scatter_cn.rst [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
 
 
 返回
 :::::::::
-返回 Task。
-
-注意
-:::::::::
-当前只支持动态图
+若为同步操作，无返回值；若为异步操作，返回``Task``，显示执行状态，并可以用于调度异步操作。
```
Comment (LiYuRio) [inline docs/api/paddle/distributed/reduce_scatter_cn.rst]: 通过task，可以查看异步操作的执行状态以及等待异步操作的结果

Comment (HermitSun) [reply docs/api/paddle/distributed/reduce_scatter_cn.rst]: done

**Review by LiYuRio — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM

**Pull request merged**